internal/pkg/consensus/election.go

```diff
@@ -41,20 +42,26 @@ func (em ElectionMachine) GenerateCandidates(poStRand []byte, sectorInfos ffi.So
 	return ep.GenerateEPostCandidates(sectorInfos, convert.To32ByteArray(poStRand), dummyFaults)
 }
 
-// GeneratePoSt creates a PoSt proof over the input PoSt candidates.  Should
+// GenerateEPoSt creates a PoSt proof over the input PoSt candidates.  Should
 // only be called on winning candidates.
-func (em ElectionMachine) GeneratePoSt(allSectorInfos ffi.SortedPublicSectorInfo, challengeSeed []byte, winners []ffi.Candidate, ep postgenerator.PoStGenerator) ([]byte, error) {
+func (em ElectionMachine) GenerateEPoSt(allSectorInfos ffi.SortedPublicSectorInfo, challengeSeed []byte, winners []ffi.Candidate, ep postgenerator.PoStGenerator) ([]byte, error) {
 	return ep.ComputeElectionPoSt(allSectorInfos, challengeSeed, winners)
 }
 
-// VerifyPoStRandomness verifies that the PoSt randomness is the result of the
+// VerifyEPoStVrfProof verifies that the PoSt randomness is the result of the
 // candidate signing the ticket.
-func (em ElectionMachine) VerifyPoStRandomness(rand block.VRFPi, ticket block.Ticket, candidateAddr address.Address, nullBlockCount uint64) error {
-	seedBuf := make([]byte, binary.MaxVarintLen64)
-	n := binary.PutUvarint(seedBuf, nullBlockCount)
-	buf := append(ticket.VRFProof, seedBuf[:n]...)
+func (em ElectionMachine) VerifyEPoStVrfProof(ctx context.Context, base block.TipSetKey,
+	epoch abi.ChainEpoch, miner address.Address, worker address.Address, vrfProof block.VRFPi) error {
+	entropy, err := encoding.Encode(miner)
+	if err != nil {
+		return errors.Wrapf(err, "failed to encode entropy")
+	}
+	randomness, err := em.chain.SampleChainRandomness(ctx, base, acrypto.DomainSeparationTag_ElectionPoStChallengeSeed, epoch, entropy)
+	if err != nil {
+		return errors.Wrap(err, "failed to generate epost randomness")
+	}
 
-	return crypto.ValidateBlsSignature(buf[:], candidateAddr, rand)
+	return crypto.ValidateBlsSignature(randomness, worker, vrfProof)
```

Comment:
wfm. Note that the actual randomness value is the digest but checking this is fine, so long as the digest is what is passed to the proofs code for proof verification